### DISCUSSION TITLE
Key repeat with `useTVEventHandler`

### DISCUSSION THREAD

**Discussion by evoactivity** [react-native-tvos/react-native-tvos#728] [Q&A] [answered]
When using the `TVFocusGuideView` if I hold an arrow key the event repeats and I move through all the focusables in my screen, but if I am using `useTVEventHandler` then I see a single `longPress` event on keydown and another `longPress` on keyup. 

How do I repeat the keydown instead of seeing a `longPress` so I get multiple `up/right/down/left` events instead. Or do I use a `setInterval` on my handler for a longPress and repeat it that way instead?

**Comment by douglowder** [comment]
@evoactivity is this on Apple TV or Android TV? Which version of RNTV are you using?

It is intentional that you should get two `longSelect/longUp/longDown` etc. events (one on start and one on end) when holding down the center button or arrow keys.

**Comment by evoactivity** [comment]
Android TV (though I will be working on an iOS build once I have android finished).
react-native-tvos@0.73.7-1

An example of the UI I'm trying to acheive would be, for example, a video seeker/scrubber so when I hold down an arrow key it moves through the track without needing to press multiple times.

I do think an approach using setInterval when detecting a `longPress` and cancelling on the next `longPress` would make sense (and opens up more options like increasing the speed of the repeat over time), but I do think for simplicity, having repeated events would be nice and match how it _seems_ to work when holding an arrow when moving through `<Pressable>`s in a `TVFocusGuideView`.

**Reply by douglowder** [reply]
The `longSelect`, `longRight`, etc. events have an `eventKeyAction` property that will be 0 for the start event, and 1 for the end event, so you can distinguish them correctly and implement features like this.

**Reply by xam60** [reply]
@douglowder Is it possible to have the same long events for other key ? (rewind / fastForward for example)
Is it possible to enable repeated key events easily (for current unsupported keys) ?

**Reply by xam60** [reply, marked as answer]
I found the answer on how to enable repeated key events. In android\app\src\main\java\...\MainApplication.kt, add at the beginning :
import com.facebook.react.config.ReactFeatureFlags

and after SoLoader.init(this, false) :
ReactFeatureFlags.enableKeyDownEvents = true
